src/crawlee/request_loaders/_sitemap_request_loader.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import asyncio
+from logging import getLogger
+from typing import TYPE_CHECKING, Any
+
+from crawlee import Request
+from crawlee._utils.docs import docs_group
+from crawlee._utils.globs import Glob
+from crawlee._utils.sitemap import ParseSitemapOptions, SitemapSource, SitemapUrl, parse_sitemap
+from crawlee.request_loaders._request_loader import RequestLoader
+
+if TYPE_CHECKING:
+    import re
+    from collections.abc import Sequence
+
+    from crawlee.storage_clients.models import ProcessedRequest
+
+
+logger = getLogger(__name__)
+
+
+@docs_group('Classes')
+class SitemapRequestLoader(RequestLoader):
```

Comment:
So the class doesn't have any automatically persisted state, right? Can you make a follow up issue to implement that?

Reply:
https://github.com/apify/crawlee-python/issues/1269